luaclang.c

```diff
@@ -104,6 +106,111 @@ static int cursor_getspelling(lua_State *L)
+        case CXCursor_TypedefDecl: 
+                return "TypedefDecl";
+        case CXCursor_IntegerLiteral: 
+                return "IntegerLiteral";
+        default: 
+                return "Unaddressed";
+    }
+}
+
+/*      
+        Format - cur:getKind()
+        Parameter - cur - Cursor whose kind is to be obtained    
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__MANIP.html#ga018aaf60362cb751e517d9f8620d490c
+*/
+static int cursor_getkind(lua_State *L)
+{
+        CXCursor *cur;
+        to_object(L, cur, CURSOR_METATABLE, 1);
+        lua_pushstring(L, cursor_kind_str(clang_getCursorKind(*cur)));
+        return 1;
+}
+
+/* 
+        Visitor function for visit_children
+        A value of this enumeration type should be returned by this functionr to indicate how visit_children proceed : 
+                1. CXChildVisit_Break - Terminates the cursor traversal 
+                2. CXChildVisit_Continue - Continues the cursor traversal with the next sibling of the cursor just visited, without visiting its children.
+                3. CXChildVisit_Recurse - Recursively traverse the children of this cursor, using the same visitor and client data.
+ */
+enum CXChildVisitResult visitor_function(CXCursor cursor, CXCursor parent, CXClientData client_data)
+{
+        lua_State *L = (lua_State*) client_data;
+        lua_pushvalue(L, 1);    
+        CXCursor *cur;
+        new_object(L, cur, CURSOR_METATABLE);           
+        *cur = cursor;
+        CXCursor *par;
+        new_object(L, par, CURSOR_METATABLE);
+        *par = parent;
+        if (lua_pcall(L, 2, 1, 0) != 0) {
+                lua_settop(L, 0);
+                lua_pushstring(L, "Function call unsuccessfull");
+                return CXChildVisit_Break;
+        }
+        const char *result = lua_tostring(L, -1);
+        if (strcmp(result, "continue") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Continue;
+        }
+        else if (strcmp(result, "recurse") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Recurse;
+        }
+        else if (strcmp(result, "break") == 0) {
+                lua_pop(L, 1);
+                return CXChildVisit_Break;
+        }
+        else {
+                lua_settop(L, 0);
+                lua_pushstring(L, "Undefined return to visitor");
+                return CXChildVisit_Break;
+        }
+}
+
+/*      
+        Format - cur:visitChildren(visitor_function)
+        Parameter - cur - Cursor whose children are to be visited   
+        More info - https://clang.llvm.org/doxygen/group__CINDEX__CURSOR__TRAVERSAL.html#ga5d0a813d937e1a7dcc35f206ad1f7a91
+        Returns nothing
+*/
+static int visit_children(lua_State *L)                               
+{
+        CXCursor *cur;
+        to_object(L, cur, CURSOR_METATABLE, 1);
+        luaL_argcheck(L, lua_isfunction(L, 2) == 1, 1, "arg  is not a function");
```

Comment:
There is a more specialised function that we could use:
```suggestion
        luaL_checktype(L, 2, LUA_TFUNCTION);
```